### ISSUE TITLE
Create Table: referrer

### ISSUE THREAD

**Issue by fyliu** [hackforla/peopledepot#68]
### Overview

We need to create the referrer table so that we can update a shared data store across hackforla.org, vrms, civictechjobs, and tables (onboarding) project.

#### Details

A table and a model are the same thing

### Action Items

- [x] identify and document table description (see spreadsheet under Resources)
  - [ ] if not, reach out to PD leads
- [x] compare and check off the data fields (below) against the ERD. Note any fields in the ERD that's missing from the list. (see Resources)
- [x] compare the check off associated tables (below) against the ERD. Note any associated tables in the ERD that's missing from the list (see Resources)
- [x] create a single model in Django (defining schema)
- [x] write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project).
- [x] write an API end point
- [x] write API unit tests
- [x] document the endpoint
- [x] uncomment `referred_by` field in User model when this issue's table is created.
- Check to see if open issues in `FK status` below
   - [ ] If all Issues listed are closed (or there are no FKs), then skip the next 3 steps
      - [x] comment out the code after you create it (see FK Status below for open/closed state)
      - [x] add an action item on the open issue to uncomment the code line you commented out when that issue's table is created. 
      - [x] Provide your file and line number as a permalink under resources in that issue

### Resources/Instructions

- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint
- See the [wiki page](https://github.com/hackforla/peopledepot/wiki/Create-Table-issues-data-gathering-workflow) for instructions.

### Description

List of referrers

### Data Fields

Name | Type | Required | FK Table | FK table Issue(s)
-- | -- | -- | -- | --
name | varchar() | TRUE | | 
url | url varchar |  | | 
referrer_type_id | int | TRUE | referrer_type | #69
contact_name | varchar |  | | 
contact_email | email varchar |  | | 

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None
   - 
### FK status
- #69 

### Associated Tables

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] user (one-to-many)
   - [x] referrer_type (many-to-one) #69 

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

**Comment by shmonks** [comment]
@fyliu Is this issue's dependency on #69 simply because it contains the FK used in the 'referrer' table? If so, I will edit this to give instructions on dealing with FKs when the other table has not yet been created, and move into New Issue Review.

**Comment by fyliu** [comment]
> [@fyliu](https://github.com/fyliu) Is this issue's dependency on [#69](https://github.com/hackforla/peopledepot/issues/69) simply because it contains the FK used in the 'referrer' table?

Yes. That's the only reason.

**Comment by shmonks** [comment]
@Neecolaa - In the ERD, this table doesn't have an FK, which I guess is why there's a question mark beside it in the top comment. Could you edit to confirm/clarify? Thanks.

**Comment by Neecolaa** [comment]
Yes, `referrer_type_id` is an FK for the table `referrer_type`. I've edited the ERD, spreadsheet, and this issue accordingly.

**Comment by shmonks** [comment]
@Neecolaa and @ExperimentsInHonesty - which fields are optional in this table?

**Comment by ExperimentsInHonesty** [comment]
### Required

- name
- referrer_type_id

### Optional
- url	
- contact_name	
- contact_email

**Comment by shmonks** [comment]
Hi @del9ra, since you'll be working on TDM this month, I'm going to unassign you from this issue and put it back into the Prioritized Backlog so you can pick it up again in May if you like
